webrecorder/browsertrix
# feat: Localize dates, bytes, and numbers

Resolves https://github.com/webrecorder/browsertrix/issues/2143

## Changes
- Defaults to user's browser language preference when displaying dates, bytes, and numbers (with caveats.) This fixes an issue where numbers were always formatted in English.
- Shows user's browser language preferences in language dropdown
- Fixes timezone not displayed in archived item detail start and finish times
- Standardizes number formatting
- Sets timezone that unit tests run in

## Manual testing

Tested with Firefox and Chrome

1. Set language preference in browser of choice
2. Log in
3. Verify dates, bytes, and numbers are displayed correctly in chosen language
4. Go to Account Settings
5. Change language preference to a language that displays numbers differently
6. Go back to org dashboard. Verify date/byte/numbers are displayed correctly

## Screenshots

| Page | Image/video |
| ---- | ----------- |
| Account Settings | <img width="1024" alt="Screenshot 2024-11-12 at 3 22 51 PM" src="https://github.com/user-attachments/assets/6fcabef9-c278-43d6-ab3f-d3d18e2a4a58"> |
| Collections | (Browser language set to Korean, no app language set)<br/><img width="1292" alt="Screenshot 2024-11-12 at 3 22 07 PM" src="https://github.com/user-attachments/assets/8a264713-66ec-4a40-b56b-953f93f74796"> |
| Archived Item Detail | <img width="200" alt="Screenshot 2024-11-12 at 3 44 13 PM" src="https://github.com/user-attachments/assets/c56862d1-877b-4b88-8321-8f70c586b26e"> |
